src/main.ts

```diff
@@ -1242,8 +1242,18 @@ export class ProfilerFeature extends SqlOpsFeature<undefined> {
 			);
 		};
 
-		let pauseSession = (sessionId: string): Thenable<boolean> => {
-			return undefined;
+		let pauseSession = (ownerUri: string): Thenable<boolean> => {
+			let params: types.PauseProfilingParams = {
+				ownerUri
+			};
+
+			return client.sendRequest(protocol.PauseProfilingRequest.type, params).then(
+				r => true,
```

Comment:
should you return the response rather than just true? It appears the response can fail with an error message which would be useful to have.

Reply:
It would depend on the behavior in the tools service. If its impossible for the response to "fail" but still come over the pipeline as a success, then this is fine, but I wouldn't understand the point of having a complex object as a response object. But if its possible for the pause to fail, and that failure will come over in the normal response, the front end will need to know that and therefore just returning true will mask any failures.

Reply:
I changed this so that pause and stop don't have complex arguments as response objects, because after checking sqltoolsservice all failures are shown as errors, never as normal response objects. I have some future work to do on the response objects for starting a session, but that will be a separate PR.